### ISSUE TITLE
Example exhaustiveness

### ISSUE THREAD

**Issue by gtrefs** [jlink/jqwik#122]
## Testing Problem
Given the following Example
```
@Example
public void exhausted_after_first_try(){
  Assume.that(false);
}
```
This causes an `AssertionFailedError` with an error message of `exhausterd after [1] tries and [1] rejections`. However, if this is an example, I would expect that it is just skipped rather than a test failure. 

## Suggested Solution

A `PropertyExecutionResult` with status `Aborted` and a `TestAbortedException` as `throwable`. The example is then just skipped.

**Comment by jlink** [comment]
Since this would require special handling for examples - which are currently just equivalent to `@Property(tries=1)` - I just want to clarify your use case cannot be solved differently: Do you need a `@Disable` feature with runtime conditions? Or is anything else important for you?

**Comment by gtrefs** [comment]
I am currently writing an extension using Lifecycle Hooks. In one test, I wanted to trigger an Exception without failing a Test. I thought that Assume.that would do it. So, the test should be disabled per se but more like skipped because the Assumption is false. But maybe I misunderstand what an Assumption is in Jqwik.

**Comment by jlink** [comment]
Which hook are you using? 

If you want to skip a property sometimes `SkipExecutionHook` might be helpful.

Alternatively `AroundPropertyHook.aroundProperty(..)` could return a result with status `ABORTED`.

**Comment by gtrefs** [comment]
I will push some code later and link it here. I am already using the SkipExecutionHook and the AroundPropertyHook.

**Comment by gtrefs** [comment]
@jlink I created PR at the testcontainers repository to include jqwik support: https://github.com/testcontainers/testcontainers-java/pull/3298 Look at this test: https://github.com/testcontainers/testcontainers-java/pull/3298/files#diff-c198c8df57d5e5517d41c314fa5c4f01

**Comment by jlink** [comment]
Check I understand that correctly: If a container cannot be started you want the test/example/property to fail silently (marked as disabled). Is that right?

Does the Jupiter extension work like that? I'm asking because I'd consider that to be a failure. Unless of course `Testcontainers.disabledWithoutDocker` has been set to true (but I think that's covered somewhere else).

**Comment by jlink** [comment]
After looking more closely at the problem and comparing with Jupiter's behaviour in similar cases your initial suggestion may indeed be the best option. An alternative is to have different API methods for skipping tests and for constraining property data. That would mean however that the term "assume" gets even more overloaded and that assumptions from assertion libs like AssertJ could no longer be used for constraining data generation.

I'll go with skipping examples if an assumption fails, i.e. if a TestAbortedException is thrown.

BTW, JUnit 4's `Assume` class uses its own `AssumptionViolationException` which is not handled by jqwik.

**Comment by gtrefs** [comment]
I updated the testcontainers hook. Containers can now be restarted between each try.

**Comment by jlink** [comment]
Fixed.

Available from Maven in 1.3.7-SNAPSHOT.
